operator/controllers/kyma_controller.go

```diff
@@ -191,18 +190,19 @@ func (r *KymaReconciler) HandleInitialState(ctx context.Context, kyma *v1alpha1.
 
 func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alpha1.Kyma) error {
 	logger := log.FromContext(ctx)
-	logger.Info("processing " + kyma.Name)
 
 	var err error
-	var modules parsed.Modules
+	var modules common.Modules
 	// these are the actual modules
 	modules, err = r.GenerateModulesFromTemplate(ctx, kyma)
 	if err != nil {
 		return r.UpdateStatusFromErr(ctx, kyma, v1alpha1.StateError,
 			fmt.Errorf("error while fetching modules during processing: %w", err))
 	}
 
-	statusUpdateRequiredFromCreation, err := r.CreateOrUpdateModules(ctx, kyma, modules)
+	sync := sync.New(r)
+
+	statusUpdateRequiredFromModuleSync, err := sync.Sync(ctx, kyma, modules)
```

Comment:
maybe a more specific name like `SyncModule`? align with `SyncModuleInfo`

Reply:
the package is already called `module/sync` so I think it would be enough, if you dont think this is the case, we can rename it.

Reply:
tbh I want to observe the meaning of the method directly from name, but not through package folder, especially here is an sub package. e.g: like the `parse.GenerateModulesFromTemplates` you defined is very expressive although it's also under `module`, but anyway, just a small habit concern.